python/sglang/srt/models/llama.py

```diff
@@ -54,6 +56,15 @@
 logger = logging.getLogger(__name__)
 
 
+USE_PRESHARDED_WEIGHTS = (os.getenv("USE_PRESHARDED_WEIGHTS") or "0") == "1"
```

Comment:
this is a bit ugly, but so far we don't seem to have a good way of plumbing through `use_presharded_weights` from server args. Any idea?

Reply:
One potential way may be creating a `WeightLoaderConfig` (say, dataclass with field `use_presharded_weights`). Then, pass around this config, from `ModelLoader` into `weight_loader`.

If we want to ensure third party weight loader compatibility, maybe we can extract logic like `call_weight_loader`, which checks whether the weight_loader function has parameter `config`, and decide whether to pass in the config. Then, old weight loaders w/o the config param can still be used.

Other brainstorms: Add fields to the `weight` tensor (by subclassing Tensor etc); use context manager - indeed temporary global variables; etc.

Reply:
Makes sense. what do you think about this plan? https://sgl-fru7574.slack.com/archives/C07PEP77X6F/p1739731214073159?thread_ts=1738963030.474379&cid=C07PEP77X6F